controller/task_queue/manager.py

```diff
@@ -0,0 +1,76 @@
+from typing import Any, List, Dict, Tuple, Callable
+
+from submodules.model import enums, Record
+
+
+import os
+from submodules.model.business_objects import task_queue as task_queue_db_bo
+from submodules.model.models import TaskQueue as TaskQueueDBObj
+from .handler import (
+    embedding as embedding_handler,
+    information_source as information_source_handler,
+    tokenization as tokenization_handler,
+    attribute_calculation as attribute_calculation_handler,
+)
+import copy
+
+from controller.task_queue import task_queue
+
+
+def add_task(
+    project_id: str,
+    task_type: enums.TaskType,
+    user_id: str,
+    task_info: Dict[str, str],
+    priority: bool = False,
+) -> str:
+    task_item = task_queue_db_bo.add(
+        project_id, task_type, user_id, task_info, priority, with_commit=True
+    )
+
+    add_task_to_task_queue(task_item)
+    return str(task_item.id)
```

Comment:
It could be that the `task_item` is deleted in the function call of `add_task_to_task_queue`. If nothing is in the queue the task is started directly and deleted if some preconditions are not fulfilled. In this case, we run into an error when accessing `task_item.id`.

Reply:
stored beforehand to prevent error